source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -58,7 +58,10 @@ DnsCacheImpl::DnsCacheImpl(
     // cache to load an entry. Further if this particular resolution fails all the is lost is the
     // potential optimization of having the entry be preresolved the first time a true consumer of
     // this DNS cache asks for it.
-    startCacheLoad(hostname.address(), hostname.port_value(), false);
+    const std::string host =
+        DnsHostInfo::normalizeHostForDfp(hostname.address(), hostname.port_value());
```

Comment:
Do we need a runtime guard for this, since it looks like a behavior change? (Or is there some reason we're safe here)

Reply:
I was debating this myself, and could be convinced of adding a runtime guard.  I just figured that no one is using this currently (preresolved hostnames config), and in any case, all that will happen if the cache key change causes a cache miss is that the DNS record will have to be fetched (which is actually what was happening previously on "production" use cases, i.e. my traffic director integration test, without us knowing about it, you can just tell from the logs).

Reply:
I think I'd be inclined to add a guard (which will default to true) since it's easy to add and better safe than sorry. I think we know that nobody is using the mobile APIs for this config, but I'm not sure if we know about DFP users in the wild?